frontend/app/studentView/components/student-team-view-tab.tsx

```diff
@@ -0,0 +1,304 @@
+}
+
+export interface TeamMembership {
+    name: string;
+    role: TeamMembershipRole;
+}
+
+export interface UserMembership {
+    team: Team;
+    role: TeamMembershipRole;
+}
+
+export interface Team {
+    id: string;
+    name: string;
+    members: TeamMembership[];
+}
+
+
+export default function StudentTeamViewTab(props: StudentTeamViewTabProps){
+    const [jwt, setAndStoreJwt] = useContext(JwtContext);
+
+    const userName = "seb";
+    const {courseSlug} = props;
+    const [teams, setTeams] = useState<Team[]>([]);
+
+    const [userMembership, setUserMembership] = useState<UserMembership | undefined>(undefined);
+
+    async function fetchTeams() {
+        try {
+            //eventually call API here
+            const res = await fetchApi(jwt, setAndStoreJwt, `teams/get/${courseSlug}`, "GET");
+            const data = await res.json();
+            const returnedTeams : Team[] = [];
+            setUserMembership(undefined);
+            data.forEach(team => {
+                const newTeam = team as Team;
+                const userMember = newTeam.members.find(m => m.name == userName);
+                if(userMember != undefined){
+                    setUserMembership({team: newTeam, role: userMember.role});
+                }
+                returnedTeams.push(team as Team);
+            })
+            console.log(JSON.stringify(returnedTeams, null, 2));
+            setTeams(returnedTeams);
+
+        } catch (error) {
+            console.error("Failed to retrieve teams", error)
+        }
+    }
+
+    useEffect(() => {
+        fetchTeams()
+    }, [courseSlug])
+
+    const memberColumnTemplate = (team : Team) => {
+        return (
+        <TeamMemberList showLabel={false} members={team.members} />)
+    }
+
+    const joinTeam = ({team} : {team: Team}) => {
+        fetchApi(jwt, setAndStoreJwt, `teams/join/request/`, "PATCH", {
+            team_id: team.id
+        }).then((response) => {
+            console.log(response.status);
+            fetchTeams();
+        })
+        //TODO
+    }
+
+    const actionsColumnTemplate = (team: Team) => {
+        if(team.id === userMembership?.team.id){
+            return (<LeaveTeamButton membership={userMembership} leaveTeam={leaveTeam}/>)
+        }
+        return (
+            <Button size="small" label="Join Team" raised text onClick={() => joinTeam({team})}/>
+        )
+    }
+
+    const [globalFilterValue, setGlobalFilterValue] = useState<string>('');
+    const [filters, setFilters] = useState<DataTableFilterMeta>({
+        global: {value: null, matchMode: FilterMatchMode.CONTAINS}
+    })
+
+    const onGlobalFilterChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const value = e.target.value;
+        let _filters = { ...filters };
+
+        // @ts-ignore
+        _filters['global'].value = value;
+
+        setFilters(_filters);
+        setGlobalFilterValue(value);
+    };
+
+    const createTeam = () => {
```

Comment:
I think autogenerated names should be created by the backend 

If someone else creates a team while I have this page open, and then i create a team, it will fail due to that team name already existing in the db